torch/_export/serde/serialize.py

```diff
@@ -474,13 +577,237 @@ def serialize(self, exported_program: ep.ExportedProgram) -> Tuple[ExportedProgr
+        elif type_ == Argument.fields().as_scalar_type.name:
+            return _SERIALIZE_TO_TORCH_DTYPE[value.as_scalar_type]
+        elif type_ == Argument.fields().as_memory_format.name:
+            return _SERIALIZE_TO_TORCH_MEMORY_FORMAT[value.as_memory_format]
+        elif type_ == Argument.fields().as_layout.name:
+            return _SERIALIZE_TO_TORCH_LAYOUT[value.as_layout]
+        elif type_ == Argument.fields().as_device.name:
+            return deserialize_device(value.as_device),
+        elif type_ == Argument.fields().as_bool.name:
+            return value.as_bool
+        elif type_ == Argument.fields().as_bools.name:
+            # convert from serialized.python.types.List to python list
+            return list(value.as_bools)
+        else:
+            raise SerializeError("Unhandled argument type:", type_)
+
+    def deserialize_outputs(self, serialized_node: Node, fx_node: torch.fx.Node) -> None:
+        # Simple case for single tensor return.
+        assert isinstance(fx_node.target, torch._ops.OpOverload)
+        if _is_single_tensor_return(fx_node.target):
+            return self.sync_serialized_node(serialized_node.outputs[0].as_tensor.name, fx_node)
+
+        # Convert multiple return types to FX format.
+        # In FX, each node only returns one value. So in order to represent
+        # multiple return values, we have to emit a `getitem` node for each
+        # return value.
+        # This performs the inverse mapping of the `serialize_outputs` call in
+        # serialization, see [NOTE: Multiple outputs]
+        output_names = []
+        if len(serialized_node.outputs) == 1:
+            assert serialized_node.outputs[0].type == Argument.fields().as_tensors.name
+            output_names = [arg.name for arg in serialized_node.outputs[0].as_tensors]
+        else:
+            for output in serialized_node.outputs:
+                assert output.type == Argument.fields().as_tensor.name
+                output_names.append(output.as_tensor.name)
+
+        for idx, name in enumerate(output_names):
+            individual_output = self.graph.create_node(
+                "call_function",
+                operator.getitem,
+                (fx_node, idx),
+                name=name,
+            )
+            self.sync_serialized_node(name, individual_output)
+            # The derived `getitem` nodes should have the same stacktrace as the
+            # original `fx_node`
+            individual_output.meta.update(deserialize_metadata(serialized_node.metadata))
+
+        # also update the metaval for `fx_node` to be a list(meta)
+        fx_node.meta["val"] = [self.serialized_name_to_meta[name] for name in output_names]
+
+
+class ExportedProgramDeserializer:
+    def __init__(self, expected_opset_version: Optional[Dict[str, int]] = None):
+        self.expected_opset_version: Dict[str, int] = (
+            {} if expected_opset_version is None else expected_opset_version
```

Comment:
ok so this hasn't been used. I can work on top of your PR to add upgrader support.